### ISSUE TITLE
[Schema Parser?] Extending root operations in schemas is not working if "# import" is used

### ISSUE THREAD

**Issue by eddeee888** [dotansimha/graphql-code-generator#2745]
## Problem
When extending schemas, the `# import ...` at the top of my second schema causes the generation process to fail. This happens for at least `fragment-matcher` and `typescript-graphql-files-modules` plugins.

## Example
For example, I have 2 schemas: `schema.graphql` and `projects.graphql`:

1. My main schema - `schema.graphql`:
```graphql
type Query
type Mutation
type Subscription
```

2. My second schema -`projects.graphql`:
```graphql
# import * from 'schema.graphql'

extend type Query {
    project(id: ID!): Project!
}

extend type Mutation {
    createProject(name: String!): Project!
}

extend type Subscription {
    projectChanges: [Project!]!
}

type Project {
    id: ID!
    title: String!
    description: String!
}
```

3. My `codegen.yml` config file (order of the schemas doesn't seem to matter):
```yaml
overwrite: true
schema: 
  - projects.graphql
  - schema.graphql
  
generates:
  src/webpack/__generated__/types.ts:
    plugins:
      - typescript
      
  src/webpack/__generated__/introspection-result.ts:
    plugins:
      - fragment-matcher

  src/webpack/__generated__/operations.d.ts:
    documents:
      - src/webpack/**/*.graphql
    plugins:
      - typescript-graphql-files-modules
```

Running codegen will give me this error:
```
Error: Type Query must define one or more fields.

    Type Mutation must define one or more fields.

    Type Subscription must define one or more fields.
        at assertValidSchema (/sandbox/node_modules/graphql/type/validate.js:80:11)
        at assertValidExecutionArguments (/sandbox/node_modules/graphql/execution/execute.js:146:35)
        at executeImpl (/sandbox/node_modules/graphql/execution/execute.js:67:3)
        at Object.execute (/sandbox/node_modules/graphql/execution/execute.js:62:35)
        at Object.exports.plugin (/sandbox/node_modules/@graphql-codegen/fragment-matcher/dist/commonjs/index.js:22:
43)
        at Object.executePlugin (/sandbox/node_modules/@graphql-codegen/core/dist/commonjs/execute-plugin.js:32:41)
```

Removing the `# import ...` statement works! But it should not fail the first place I think 🤔 

You can see this behaviour here: [example codesandbox](https://codesandbox.io/s/gcg-18-experiments-v2-uq5he)

Running `yarn gg` should show the errors. Removing the `# import ...` in `projects.graphql` and it works!

## Expected behavior

It should work with or without `# import ...` statements

## Environment
- "@graphql-codegen/cli": "1.8.1"
- "@graphql-codegen/fragment-matcher": "1.8.1"
- "@graphql-codegen/typescript-graphql-files-modules": "^1.8.1"
- "@graphql-codegen/typescript": "1.8.1" ( This actually works! I wonder what the difference is? )
- NodeJS: v10.16.0

**Comment by dotansimha** [comment]
@eddeee888 When using `#import` syntax, it uses `graphql-import` for building the typedefs and schema. When you removed it, it used `graphql-toolkit` to merge the typedefs (which does it in a different way, so that's probably the reason for that). 
I think, that if you can use it without the `#import` - just go for it, anyway the `graphql-import` doesn't have added value in this use case, because IDE will behave the same, and you still need to load your schema in your server somehow. 
In terms of codegen, it works better with `graphql-toolkit` merging logic, and it's more graceful in terms of syntax and conflicts.

**Comment by eddeee888** [comment]
@dotansimha thanks for the detailed response! For our use case, we have decided to use the server graphql endpoint to generate code. 

Just out of curiosity, why does this library need both `graphql-import` and `graphql-toolkit` to parse a file? It's also interesting that `graphql-import` states that it's using a custom way to import schema [here](https://oss.prisma.io/content/graphql-import/overview#description) so I guess not using `# import ...` is better in either case

**Comment by dotansimha** [comment]
Actually, we added support for `graphql-import` only after we had `graphql-toolkit` (which was created from functions that was originally inside the codegen). 
Now we are the maintainers of `graphql-import` so the logic of it might move into `graphql-toolkit` and we can have it supported for all needs.

**Comment by ardatan** [comment]
@eddeee888 Could you try with this canary version?
We changed `graphql-import`'s behavior and fixed extension issue 
`1.10.1-alpha-7abaacf0.6`

**Comment by dotansimha** [comment]
Fixed in 1.11.0
